### DISCUSSION TITLE
Errore di non autorizzazione 401 lista clienti

### DISCUSSION THREAD

**Discussion by piccio85** [fattureincloud/api#396] [Q&A] [answered]
Describe the bug
Sviluppo un software per diversi clienti che utilizzando anche fatture in cloud
Dal 9 di gennaio 2 miei clienti (su una trentina) che usano le API con fatture in cloud quando provano ad accedere alla lista clienti ottengo un errore "401 non autorizzato".

Il software quindi e' scritto correttamente in quanto molti miei clienti riescono a lavorare correttamente.

Env info
PHP 8.0.30
PHP SDK 2.0.24

To Reproduce
Eseguo una chiamata per avere la lista dei clienti ma ottengo un errore 401

```php

$companies = $userInstance->listUserCompanies();
if (empty($companies->getData()->getCompanies())){
	return;
}
$companyId = $companies->getData()->getCompanies()[0]->getId();

$cont_pagina = 1;
$listaClienti = [];

try {
	$resClienti = $clientInstance->listClients($companyId, null, null, null, $cont_pagina, 100);
} catch (Exception $e) {
	return;
}
``` 

401] Client error: GET https://api-v2.fattureincloud.it/c/944984/entities/clients?page=1&per_page=100 resulted in a 401 Unauthorized response:
{"error":"unauthorized","error_description":"The given user is not authorized to use this application."}

Mostro alcuni screenshot:
![image](https://github.com/fattureincloud/api/assets/12586165/58c5256d-7f2d-4543-a85e-ac58150eeeed)

![image](https://github.com/fattureincloud/api/assets/12586165/011c4816-3959-445b-b35c-cabe42d94cf0)

![image](https://github.com/fattureincloud/api/assets/12586165/266743d9-c4c1-4836-9453-ae87e2f0ceda)

**Comment by emajo** [comment, marked as answer]
Ciao @piccio85 dall'errore _The given user is not authorized to use this application_ sembrerebbe che gli utenti in questione non siano stati aggiunti alla whitelist dell'applicazione, più informazioni [qui](https://developers.fattureincloud.it/docs/basics/create-an-app/#lock-private-apps) (sezione private apps).

Grazie,
Emanuele.

**Reply by valmoz** [reply]
Come discusso via mail, la soluzione consiste nell'utilizzare un'unica app pubblica per tutti gli utenti.
Non si dovrebbe mai chiedere agli utenti di creare un'app dalla sezione sviluppatore.

Grazie
Mauro
